sources/packages/backend/apps/queue-consumers/src/processors/schedulers/student-application-notifications/_tests_/student-application-notifications.scheduler.e2e-spec.ts

```diff
@@ -293,5 +294,130 @@ describe(
+            },
+          },
+        );
+
+        // Queued job.
+        const mockedJob = mockBullJob<void>();
+
+        // Act
+        await processor.processQueue(mockedJob.job);
+
+        // Assert
+        expect(
+          mockedJob.containLogMessages([
+            `Second disbursements with pending status that generated notifications: ${application.currentAssessment.id}`,
+          ]),
+        ).toBe(true);
+        const notification = await db.notification.findOne({
+          select: {
+            id: true,
+            messagePayload: true,
+          },
+          relations: { notificationMessage: true },
+          where: {
+            notificationMessage: {
+              id: NotificationMessageType.StudentSecondDisbursementNotification,
+            },
+            dateSent: IsNull(),
+            user: { id: student.user.id },
+          },
+        });
+        expect(notification).toBeDefined();
+        expect(notification.messagePayload).toStrictEqual({
+          email_address: application.student.user.email,
+          template_id: "55fcf228-b899-49a7-ab80-9b854c0bd884",
+          personalisation: {
+            lastName: application.student.user.lastName,
+            givenNames: application.student.user.firstName,
+          },
+        });
+      },
+    );
+
+    it(
+      "Should not generate a second disbursement reminder notification for a student " +
+        "when a notification is already sent for the second disbursement still pending.",
+      async () => {
+        // Arrange
+        const student = await saveFakeStudent(db.dataSource);
+        // Create an application with the second disbursement still pending with Required COE
+        // status and disbursement date has passed.
+        const application = await saveFakeApplicationDisbursements(
+          db.dataSource,
+          { student },
+          {
+            applicationStatus: ApplicationStatus.Completed,
+            createSecondDisbursement: true,
+            firstDisbursementInitialValues: {
+              coeStatus: COEStatus.completed,
+              disbursementScheduleStatus: DisbursementScheduleStatus.Sent,
+              disbursementDate: getISODateOnlyString(addDays(-10)),
+            },
+            secondDisbursementInitialValues: {
+              coeStatus: COEStatus.required,
+              disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+              disbursementDate: getISODateOnlyString(addDays(-2)),
+            },
+          },
+        );
+        // Create a notification for assessment associated with the second disbursement.
+        const notification = createFakeNotification(
+          {
+            user: student.user,
+            notificationMessage: await db.notificationMessage.findOne({
+              where: {
+                id: NotificationMessageType.StudentSecondDisbursementNotification,
+              },
+            }),
```

Comment:
I believe there is no need to execute the `db.notificationMessage.findOne` to get the notification from DB since its ID is known. Can you please try the below?

```ts
const notification = createFakeNotification(
  {
    user: student.user,
    notificationMessage: {  id: NotificationMessageType.StudentSecondDisbursementNotification } as NotificationMessage,
  },
  {
    initialValue: {
      metadata: { assessmentId: application.currentAssessment.id },
    },
  },
);
```